boat_simulator/nodes/physics_engine/kinematics_computation.py

```diff
@@ -0,0 +1,125 @@
+"""This module contains the kinematics computations for the boat."""
+
+from typing import Tuple
+from numpy.typing import ArrayLike
+from boat_simulator.common.types import Scalar
+from boat_simulator.nodes.physics_engine.kinematics_data import KinematicsData
+import numpy as np
+import boat_simulator.common.constants as constants
+
+
+class BoatKinematics:
+    # Private class member default
+    __relative_data = KinematicsData()
+    __global_data = KinematicsData()
+
+    def __init__(self, timestep: Scalar, mass: Scalar, inertia: ArrayLike) -> None:
+        self.__timestep = timestep
+        self.__boat_mass = mass
+        self.__inertia = np.array(inertia, dtype=np.float32)
+        self.__inertia_inverse = np.linalg.inv(self.__inertia)
+
+    # Updates the kinematics data and returns the kinematics data objects
+    def step(self, net_force: ArrayLike, net_torque: ArrayLike
+             ) -> Tuple[KinematicsData, KinematicsData]:
+
+        next_relative_acceleration, next_relative_velocity = \
+            self.__update_linear_relative_data(net_force)
+
+        yaw_radians = self.__update_ang_data(net_torque)
+
+        self.__update_linear_global_data(
+            next_relative_acceleration, next_relative_velocity, yaw_radians
+            )
+
+        return (self.relative_data, self.global_data)
+
+    def __update_linear_relative_data(self, net_force: ArrayLike) -> Tuple[ArrayLike, ArrayLike]:
+        next_relative_acceleration = self.__compute_next_relative_acceleration(
+            self.boat_mass, net_force
+            )
+        next_relative_velocity = self.__compute_next_velocity(
+            self.relative_data.linear_velocity, self.relative_data.linear_acceleration
+            )
+
+        self.__relative_data.linear_acceleration = next_relative_acceleration
+        self.__relative_data.linear_velocity = next_relative_velocity
+
+        return (next_relative_acceleration, next_relative_velocity)
+
+    def __update_linear_global_data(self, next_relative_acceleration: ArrayLike,
+                                    next_relative_velocity: ArrayLike,
+                                    yaw_radians: Scalar) -> None:
+
+        next_global_acceleration = next_relative_acceleration * \
+            np.array([np.cos(yaw_radians), np.sin(yaw_radians), 0])
+        next_global_velocity = next_relative_velocity * \
+            np.array([np.cos(yaw_radians), np.sin(yaw_radians), 0])
```

Comment:
Adding a comment saying that we neglect the z-directional velocity and acceleration will make this less confusing for new readers of this code.